src/edgepi/adc/adc_voltage.py

```diff
@@ -70,17 +69,23 @@ def code_to_voltage(code: list[int], adc_info: ADCReadInfo, calibs: CalibParam)
     Returns:
         `float`: voltage value (V) corresponding to `code`
     """
-    code_bits = bitstring_from_list(code[:adc_info.num_data_bytes])
-    num_bits = adc_info.num_data_bytes * 8
-    code_val = code_bits.uint
 
-    if _is_negative_voltage(code_bits):
+    num_bits = adc_info.num_data_bytes * 8
+    if adc_info.num_data_bytes == ADC1_NUM_DATA_BYTES:
+        code_val = combine_to_uint32(code[0], code[1], code[2], code[3])
+    elif adc_info.num_data_bytes == ADC2_NUM_DATA_BYTES:
+        code_val = combine_to_uint32(0, code[0], code[1], code[2])
+    else:
+        raise ValueError(
+            f"code has unexpected number of bytes {adc_info.num_data_bytes}, "
+            "expected 4 for ADC1 or 3 for ADC2"
+        )
```

Comment:
This looks like 106-115 below, how about abstracting it out to a helper method?

Reply:
Lovely idea! I decided to combine `code_to_voltage` and `code_to_voltage_single_ended` as they had more shared code than this snippet. It even resulted in some code improvements in functions further up the line as well.

https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/424/commits/7a9d8847b3b9e33727558a0c6ea2c8ac74cd8545